src/docserv/deliverable.py

```diff
@@ -409,7 +412,13 @@ def extract_root_id(self, command, thread_id):
         if not result:
             return False
         self.title = self.out.decode('utf-8')
+        dchash['cmd'] = "%s %s" % (os.path.join(BIN_DIR, 'docserv-dchash'), dc_path)
```

Comment:
Hmn. Why do you create a new tool instead of using the [hashlib](https://docs.python.org/3/library/hashlib.html#module-hashlib) library? :thinking:

Reply:
i had a bash implementation of what i wanted lying around ... i didn't want to reimplement in python right now.

Reply:
In case you need one day a Python implementation, here is the code:

```python
import hashlib
def sha256(filename):
    """Returns the SHA256 hexdigest from a file"""
    sha = hashlib.sha256()
    with open(filename, 'rt') as fh:
        for line in fh:
            sha.update(line.encode("utf-8"))
    return sha.hexdigest()
```

Inside Python:

```python

>>> ini = "/path/to/GH/susedoc/docserv/config/docserv.ini"
>>> sha256(ini)
'7de99afcd14856af55eeee2ea2bff28bbf985cc6d453487bf49a6c3df2722b25'
```

Comparing it with `sha256`:

```shell
sha256sum /docserv/config/docserv.ini
7de99afcd14856af55eeee2ea2bff28bbf985cc6d453487bf49a6c3df2722b25  ...
```

Reply:
your code does something different than dchash, as it doesn't filter the input lines or remove quotes/spaces and sha* is overkill given that this is not security-related functionality and we only have maybe a few hundred input metadata cache files.